Zend/zend_vm_def.h

```diff
@@ -1939,6 +1944,24 @@ ZEND_VM_C_LABEL(fetch_obj_r_no_object):
 		}
 	} while (0);
 
+	if (UNEXPECTED(!Z_VERIFIED_P(EX_VAR(opline->result.var)) && ZEND_CLASS_HAS_TYPE_HINTS(Z_OBJCE_P(container)))) {
+		zend_property_info *prop_info = zend_object_fetch_property_type_info(container, offset, ((OP2_TYPE == IS_CONST) ? CACHE_ADDR(Z_CACHE_SLOT_P(offset)) : NULL));
+
+		if (prop_info) {
+			if (Z_TYPE_P(EX_VAR(opline->result.var)) == IS_NULL) {
+				zend_throw_exception_ex(zend_ce_type_error, prop_info->type,
+					"Typed property %s::$%s must not be accessed before initialization",
+					ZSTR_VAL(prop_info->ce->name),
+					Z_STRVAL_P(offset));
+				HANDLE_EXCEPTION();
+			} else if (!zend_verify_property_type(prop_info, EX_VAR(opline->result.var), EX_USES_STRICT_TYPES())) {
+				zend_verify_property_type_error(prop_info, Z_STR_P(offset), EX_VAR(opline->result.var));
+				HANDLE_EXCEPTION();
+			}
```

Comment:
This won't work. You should prevent any ability of assignment of incorrect type.
If you don't do this, inserting additional type check only in this place won't enough.